File: examples/go/runfile.go
// Example for using runfiles with Go.
package main

import (
	"fmt"
	"log"
	"os"
	"path/filepath"

	"github.com/bazelbuild/rules_go/go/runfiles"
)

const (
	someFile = "examples/runfile.txt"
)

func main() {
	r, err := runfiles.New()
	if err != nil {
		log.Fatalf("Failed to create Runfiles object: %v", err)
	}

	root := runfiles.CallerRepository()
	if root == "" {
		root = "_main"
	}

	path, err := r.Rlocation(filepath.Join(root, someFile))
	if err != nil {
		log.Fatalf("Failed to find rlocation: %v", err)
	}

	fmt.Println("The content of my runfile is:")
	data, err := os.ReadFile(path)
	if err != nil {
		log.Fatalf("Failed to read file: %v", err)
	}
	fmt.Print(string(data))
}

File: examples/py/runfile.py
"""Example for using runfiles with Python."""

import pathlib

from python.runfiles import runfiles

SOME_FILE = pathlib.Path("examples/runfile.txt")

r = runfiles.Create()
root = r.CurrentRepository()
if not root:
  root = "_main"
realPathToSomeFile = r.Rlocation(str(root / SOME_FILE))

print("The content of the runfile is:")
with open(realPathToSomeFile, "r") as f:
  print(f.read(), end="")
